core/src/main/java/io/hyperfoil/core/generators/ObjectValueProviderBuilder.java

```diff
@@ -0,0 +1,74 @@
+package io.hyperfoil.core.generators;
+
+import io.hyperfoil.api.config.BenchmarkDefinitionException;
+import io.hyperfoil.api.config.InitFromParam;
+import io.hyperfoil.api.session.Access;
+import io.hyperfoil.api.session.Session;
+import io.hyperfoil.core.session.SessionFactory;
+import io.hyperfoil.function.SerializableToObjectFunction;
+
+public abstract class ObjectValueProviderBuilder<P, V> implements InitFromParam<ObjectValueProviderBuilder<P, V>> {
+   protected final P parent;
+   protected final V defaultValue;
+   protected V value;
+   protected String fromVar;
+
+   public ObjectValueProviderBuilder(P parent, V defaultValue) {
+      this.parent = parent;
+      this.defaultValue = defaultValue;
+   }
+
+   public P end() {
+      return parent;
+   }
+
+   /**
+    * Initialize with a constant value.
+    *
+    * @param param Constant value.
+    * @return Self.
+    */
+   @Override
+   public abstract ObjectValueProviderBuilder<P, V> init(String param);
+
+   /**
+    * Initialize with a constant value.
+    *
+    * @param value Constant value.
+    * @return Self.
+    */
+   public ObjectValueProviderBuilder<P, V> value(V value) {
+      this.value = value;
+      return this;
+   }
+
+   /**
+    * Initialize with a value from session variable.
+    *
+    * @param fromVar Variable name.
+    * @return Self.
+    */
+   public ObjectValueProviderBuilder<P, V> fromVar(String fromVar) {
+      this.fromVar = fromVar;
+      return this;
+   }
+
+   public abstract <R extends SerializableToObjectFunction<Session>> R buildFromAccess(Access access);
+
+   public abstract <R extends SerializableToObjectFunction<Session>> R buildFromConst();
+
+   public <R extends SerializableToObjectFunction<Session>> R build() {
```

Comment:
There shouldn't be any boxing/unboxing during runtime - in the `build()` itself it's fine but the concrete class must return either `SerializableToIntFunction<Session>` or `SerializableFunction<Session, String>`

Reply:
Those 3 methods were moded to the concrete implementation